glusterd2/commands/volumes/volume-start.go

```diff
@@ -104,22 +107,28 @@ func volumeStartHandler(w http.ResponseWriter, r *http.Request) {
 			UndoFunc: "vol-start.StartBricksUndo",
 			Nodes:    volinfo.Nodes(),
 		},
+		{
+			DoFunc:   "vol-start.UpdateVolinfo",
+			UndoFunc: "vol-start.UpdateVolinfo.Undo",
+			Nodes:    []uuid.UUID{gdctx.MyUUID},
+		},
 	}
 
-	if err := txn.Ctx.Set("volinfo", volinfo); err != nil {
+	if err := txn.Ctx.Set("oldvolinfo", volinfo); err != nil {
 		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
 		return
 	}
 
-	if err := txn.Do(); err != nil {
-		logger.WithError(err).WithField(
-			"volume", volname).Error("transaction to start volume failed")
+	volinfo.State = volume.VolStarted
+
+	if err := txn.Ctx.Set("volinfo", volinfo); err != nil {
 		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err)
```

Comment:
Logging.

Reply:
It's logged internally at `Error` log level.